compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/NilLiteralTest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerinalang.compiler.parser.test.syntax.expressions;
+
+import org.testng.annotations.Test;
+
+/**
+ * Test parsing nil literal.
+ */
+public class NilLiteralTest extends AbstractExpressionsTest {
+
+    // Valid source tests
+
+    @Test
+    public void testSimpleNilLiteral() {
+        test("()", "basic-literals/nil_literal_assert_01.json");
+        test("null", "basic-literals/nil_literal_assert_02.json");
+    }
+
+    @Test
+    public void testNilLiteral() {
+        test("() != null + a", "basic-literals/nil_literal_assert_03.json");
+    }
+}
```

Comment:
Shall we add some more tests, to validate the usages of nil-literal (`()`), such as var-decl, defaultable-params, in object/record fields, etc?